doc/dev/data-structures.md

```diff
@@ -362,6 +362,37 @@ synchronised.
 
 Continue reading [with the error handling](error-handling.md).
 
-## Order Preserving Minimal Perfect Hash Map
+## Order Preserving Minimal Perfect Hash Map (aka OPMPHM)
 
 All structs are defined in [opmphm.h](/src/include/kdbopmphm.h).
+
+Generally the OPMPHM is a arbitrary function.
+
+The desired return value aka the order is set in `OpmphmOrder->index.p`,
+additionally a minimum and maximum order is needed in `OpmphmInit->minOrder`
+and `OpmphmInit->maxOrder`.
+
+Due to the nature of the OPMPHM all elements need to be known at build time.
```

Comment:
Define what "build time" is, maybe give all phases a name or simply call them like the functions are named.